hibernate-core/src/main/java/org/hibernate/Hibernate.java

```diff
@@ -195,4 +195,23 @@ public static boolean isPropertyInitialized(Object proxy, String propertyName) {
 		return true;
 	}
 
+    /**
+     * Unproxies a {@link HibernateProxy}. If the proxy is uninitialized, it automatically triggers an initialization.
+     * In case null object was supplied, the return value will be null.
+     *
+     * @param proxy the {@link HibernateProxy} to be unproxied
+     * @return if proxy is not null then the proxy's underlying implementation object, null otherwise
+     */
+    public static Object unproxy(Object proxy) {
+        if (proxy == null) {
+            return null;
+        }
+        if (proxy instanceof HibernateProxy) {
+            HibernateProxy hProxy = (HibernateProxy) proxy;
+            LazyInitializer initializer = hProxy.getHibernateLazyInitializer();
+            return initializer.getImplementation();
+        } else {
+            throw new IllegalArgumentException("The proxy should be a HibernateProxy");
```

Comment:
Looks good, but I think I'll change it so that it simply returns the proxy parameter in the else branch. The reason for this change is if you want to unproxy a given object which you don't know if it was initialized or not. I've seen such a use cases when trying to initialize all proxies before returning them to the Web layer, in some applications.